core/src/style/pointStyleBuilder.cpp

```diff
@@ -138,11 +138,17 @@ auto PointStyleBuilder::applyRule(const DrawRule& _rule, const Properties& _prop
     _rule.get(StyleParamKey::angle, p.labelOptions.angle);
 
     auto sizeParam = _rule.findParameter(StyleParamKey::size);
-    if (sizeParam.stops && sizeParam.value.is<float>()) {
-        float lowerSize = sizeParam.value.get<float>();
-        float higherSize = sizeParam.stops->evalWidth(m_zoom + 1);
```

Comment:
@tallytalwar evalWidth does the exponential interpolation which is required to interpolate properly relative to zoomlevels.

say for the stops [0, 1px],[2, 4px] the result should be 2 at zoom 1, not 3 (x=(1-0)/(2-0), 1*(1-x)+4*x).